integration-tests/upstream/test-source-fixed.gradle

```
@@ -0,0 +1,71 @@
+//Tasks for "fixing" small things in upstream test sources and running them
+
+def output = file("$buildDir/test-source-fixed")
+
+sourceSets.test.java {
+  srcDir(output)
+}
+
+def fix = { String testClassName,
+            Class<TestSourceFix> taskClass ->
```

Strange to define a variable when it's a function, how about
```suggestion
void fix(String testClassName, Class<TestSourceFix> taskClass) {
```
or better yet:
```suggestion
static void fix(Project project, String testClassName, Class<TestSourceFix> taskClass) {
```
just to need to declare which variable is captured from `project`.

---

You can't capture local script variables in functions, so you'd have to pass each one(just `output` in this case) of them to the function in addition to passing `project`.

---

... I'll never understand Groovy. 😅